File: ObjectiveC/LP4_CurrencyConverter/CurrencyConverter/CurrencyConverter/CurrencyRateModel.m
//
//  CurrencyRateModel.m
//  CurrencyConverter
//
//  Created by Pope, Drewry on 4/17/17.
//  Copyright © 2017 assignmentInClass4 Drew Pope. All rights reserved.
//

#import "CurrencyRateModel.h"
NSString *const USD = @"US Dollar";
NSString *const Euro = @"Euro";
NSString *const GBP = @"British Pound";
NSString *const Yen = @"Japanese Yen";

@implementation CurrencyRateModel

+ (NSDecimalNumber *)exchangeRateFrom:(NSString *)fromCurrency to:(NSString *)toCurrency {
    NSDictionary *conversionRates = @{
                                     USD:@{Euro: [NSDecimalNumber decimalNumberWithString:@"0.94"],
                                           GBP: [NSDecimalNumber decimalNumberWithString:@"0.8"],
                                           Yen: [NSDecimalNumber decimalNumberWithString:@"108.97"]},
                                     Euro:@{USD: [NSDecimalNumber decimalNumberWithString:@"1.06"],
                                            GBP: [NSDecimalNumber decimalNumberWithString:@"0.85"],
                                            Yen: [NSDecimalNumber decimalNumberWithString:@"116.04"]},
                                     GBP:@{USD: [NSDecimalNumber decimalNumberWithString:@"1.18"],
                                           Euro: [NSDecimalNumber decimalNumberWithString:@"1.256"],
                                           Yen: [NSDecimalNumber decimalNumberWithString:@"136.903"]},
                                     Yen:@{USD: [NSDecimalNumber decimalNumberWithString:@"0.0092"],
                                           Euro: [NSDecimalNumber decimalNumberWithString:@"0.0086"],
                                           GBP: [NSDecimalNumber decimalNumberWithString:@"0.0073"]}};

    //[conversionRates valueForKeyPath: [NSString stringWithFormat:@"%@.%@", fromCurrency, toCurrency]];
    //NSLog(@"\nfromCurrency = %@\ntoCurrency = %@\n\nExchange Rate = %@", fromCurrency, toCurrency, conversionRates[fromCurrency][toCurrency]);
    return conversionRates[fromCurrency][toCurrency];
}
@end

File: Swift/LP3_currencyConverter/LP3_currencyConverter/CurrencyRateModel.swift
//
//  CurrencyRateModel.swift
//  LP3_currencyConverter
//
//  Created by Pope, Drewry on 5/8/17.
//  Copyright © 2017 assignmentInClass5 Drew Pope. All rights reserved.
//

import UIKit

class CurrencyRateModel: NSObject {
    static func exchangeRates(fromCurrency:String, toCurrency:String) -> Double {
        let conversionRates = ["US Dollar": ["Euro":0.94, "British Pound":0.8, "Japanese Yen":108.97],
                               "Euro": ["US Dollar":1.06, "British Pound":0.85, "Japanese Yen":116.04],
                               "British Pound": ["US Dollar":1.18, "Euro":1.256, "Japanese Yen":136.903],
                               "Japanese Yen": ["US Dollar":0.0092, "Euro":0.0086, "British Pound":0.0073]]
        return conversionRates[fromCurrency]![toCurrency]!
    }
}

